src/Resolver/ObjktResolver.ts

```diff
@@ -47,6 +47,30 @@ export class ObjktResolver {
 		return ctx.genTokLoader.load(objkt.issuerId)
 	}
 
+	@FieldResolver((returns) => String, {
+		description: "IPFS uri pointing to the web page hosting the Generator code.",
+	})
+	artifactUri(@Root() objkt: Objkt, @Ctx() ctx: RequestContext) {
+		if (objkt.artifactUri) return objkt.artifactUri;
+		return ctx.genTokLoader.load(objkt.metadata?.artifactUri);
+	}
+
+	@FieldResolver((returns) => String, {
+		description: "IPFS uri pointing to the 300x300 (contained) thumbnail",
+	})
+	thumbnailUri(@Root() objkt: Objkt, @Ctx() ctx: RequestContext) {
+		if (objkt.thumbnailUri) return objkt.thumbnailUri;
+		return ctx.genTokLoader.load(objkt.metadata?.thumbnailUri);
```

Comment:
The genTokLoader is a [DataLoader](https://github.com/graphql/dataloader) designed to batch queries to get a Generative Token from the database. Its load function accepts a Generative Token ID.

Ultimately, requesting the DataLoader in here is not required, as the data is already available in `objkt.metadata.thumbnailUri` already stored in memory at this point of execution.